### ISSUE TITLE
[ISSUE #179]

### ISSUE THREAD

**Issue by xiaosong372089396** [apache/rocketmq-operator#187]
After helm operator is installed, the replica can be deleted, but the pod container cannot be deleted. The reason is that the replicaSet cannot be deleted. After the next deployment, a new container is added. It is very distressing. Is there any solution?
The new deployment will be in replicaSet+1 and cannot be deleted directly through helm uninstall.

kubectl get rs 

rocketmq-operator-66c596bf54                        1         1         1       5m3s
rocketmq-operator-674f864b48                        1         1         0       11m
rocketmq-operator-c48db88cc                         1         1         0       5m59s

**Comment by caigy** [comment]
Did you mean the pods created by the operator were not removed after running `helm uninstall`? Actually the operator avoids the cascading delete operations, which seems to be high-risk in production. 
Users have to delete pods of RocketMQ itself mannually.

**Comment by xiaosong372089396** [comment]
> 您的意思是操作员创建的 Pod 在运行后没有被删除吗`helm uninstall`？实际上操作员避免了级联删除操作，这在生产中似乎是高风险的。 用户必须手动删除RocketMQ本身的pod。


No, the reason is that the rocketmq operator service installed through helm install runs normally, but the rocketmq operator container pod cannot be deleted through helm uninstall. The reason is that the rocketmq operator container in the replicaSet cannot be deleted. As a result, after helm uninstall is executed, helm install will be executed again. +1 on a one-time basis, there will be 2 rocketmq operator container services. I am very distressed. I hope it can be solved, or is there any way to solve it?

**Comment by xiaosong372089396** [comment]
> Did you mean the pods created by the operator were not removed after running `helm uninstall`? Actually the operator avoids the cascading delete operations, which seems to be high-risk in production. Users have to delete pods of RocketMQ itself manually.


No, the reason is that the rocketmq Operator service installed through helm install runs normally, but the rocketmq Operator container pod cannot be deleted through helm uninstall. The reason is that the rocketmq operator container in replicaSet cannot be deleted. As a result, after helm uninstall is executed, helm install will be executed again. One time +1, there will be 2 rocketmq operator container services. I am very troubled. Hope it can be solved, or is there any way to solve it?

kubectl get rs

Rocketmq-operator-66c596bf54 1 1 1 5m3s
Rocketmq-operator-674f864b48 1 1 0 11m
Rocketmq-operator-c48db88cc 1 1 0 5m59s

There are multiple replica

**Comment by caigy** [comment]
@xiaosong372089396 Pls also show the results of `kubectl get deploy` after running `helm uninstall`.

**Comment by RockteMQ-AI** [comment]
**Issue Evaluation**

Category: `question`

This is likely caused by finalizers or owner references preventing clean deletion. Common solutions:
1. Remove finalizers manually: `kubectl patch deployment <name> --type=json -p '[{"op":"remove","path":"/metadata/finalizers"}]'`
2. Check if the operator's webhook is still running (it may block deletion)
3. Delete the ReplicaSet directly: `kubectl delete rs <name> --force --grace-period=0`

Ensure the operator pod is not in a CrashLoopBackOff state, as this can prevent cleanup.

---
*Automated evaluation by RockteMQ-AI*
